### PR TITLE
Add `model.train()` line to ReadMe training example 

### PR DESCRIPTION
**Pull request by Santosh-Gupta**
## ❓ Questions & Help

I added `model.train()` line to ReadMe training example, at this section
https://github.com/huggingface/transformers#optimizers-bertadam--openaiadam-are-now-adamw-schedules-are-standard-pytorch-schedules

```
for batch in train_data:
    model.train()
    loss = model(batch)
    loss.backward()
    torch.nn.utils.clip_grad_norm_(model.parameters(), max_grad_norm)  # Gradient clipping is not in AdamW anymore (so you can use amp without issue)
    optimizer.step()
    scheduler.step()
    optimizer.zero_grad()
```

In the API [ https://huggingface.co/transformers/main_classes/model.html#transformers.PreTrainedModel.from_pretrained ] it says

>The model is set in evaluation mode by default using model.eval() (Dropout modules are deactivated) To train the model, you should first set it back in training mode with model.train()

And I see this is the case for every example in the 'examples' folder, such as this one

https://github.com/huggingface/transformers/blob/master/examples/run_lm_finetuning.py#L227

```
        for step, batch in enumerate(epoch_iterator):
            inputs, labels = mask_tokens(batch, tokenizer, args) if args.mlm else (batch, batch)
            inputs = inputs.to(args.device)
            labels = labels.to(args.device)
            model.train()
            outputs = model(inputs, masked_lm_labels=labels) if args.mlm else model(inputs, labels=labels)
            loss = outputs[0]  # model outputs are always tuple in transformers (see doc)
```

There is a `model.train()` in each training loop, before the loss/output line. 

Some may develop their training loops based on the readme example, inadvertently deactivating that dropout layers during training.

### CODE REVIEW

**Comment by LysandreJik** [comment]
Instead of merging 5 commits I pushed a commit on master 68f7064 of which you are an author.

**Comment by Santosh-Gupta** [comment]
Whoops, I didn't realized somehow I did 5 commits
